src/main/java/core/basesyntax/service/ShopServiceImpl.java

```diff
@@ -0,0 +1,37 @@
+package core.basesyntax.service;
+
+import core.basesyntax.dao.FruitDao;
+import core.basesyntax.domain.Fruit;
+import core.basesyntax.service.strategy.OperationStrategy;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private static final String APPLE_NAME = "apple";
+    private static final String BANANA_NAME = "banana";
+    private final OperationStrategy operationStrategy;
+    private final FruitDao fruitDao;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy, FruitDao fruitDao) {
+        this.operationStrategy = operationStrategy;
+        this.fruitDao = fruitDao;
+    }
+
+    @Override
+    public void process(List<Fruit> fruits) {
```

Comment:
The 'process' method is currently tied to processing only apples and bananas, which violates the Open/Closed Principle of SOLID. Consider refactoring this method to handle any type of fruit without modification to its code.